[ADG-5358] KnowledgeBase AdGuard for Android (#9)
AdguardTeam/KnowledgeBase: docs/adguard-for-android/solving-problems/background-work.md

---
title: How to protect AdGuard from being disabled by the system
sidebar_position: 9
---

> In some cases, apps won't stay in the background ("alive" or in a sleep mode) due to Android OS optimization function, or so called "battery save mode" — this function can kill background applications. It may be inconvenient to relaunch them each time they are getting closed. To avoid the background app termination you need to follow these steps which we described separately for each manufacturer (version) of Android OS. Note that instructions for different manufacturers are mostly very similar.

Reviewer: ```suggestion
In some cases, apps won't stay in the background ("alive" or in a sleep mode) due to the Android OS optimization function, or the so-called "battery save mode" — this function can kill background applications. It may be inconvenient to relaunch them each time they are getting closed. To avoid the background app termination, you need to follow the steps we described separately for each manufacturer (version) of Android OS. Note that instructions for different manufacturers are mostly very similar.
```

## Asus

Information on Asus devices is still far from being exhaustive so there may be more issues than listed here. We're going to update this part of the article when we know more.

Go to Settings - Power Manager - Mobile Manager - Settings - Battery-saving options - Auto-start manager - AdGuard - uncheck.

## Xiaomi

#### MIUI 12.0.8+

To let your AdGuard app run successfully in the background, do the following:

> In *Settings* > *Apps* > *Manage apps* > scroll down to locate *AdGuard*, set *Autostart* to "On".

Reviewer: где-то >, где-то –, где-то –>
давай везде одинаково сделаем

![Xiaomi Settings](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomi4en.jpeg)

Scroll down to *Battery saver*, tap it, and set to "No restrictions".
 
![Miui](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/miui12_en_0a.png)

Run the *Security* app.

Tap the *Gear* icon at the top-right corner of the screen.

![Miui Settings](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/miui12_en_1.jpeg) 

> Tap *Boost speed* in Feature Settings

Reviewer: точка

![Miui Settings](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/miui12_en_2.png)   

Tap *"Lock apps"*.

![Miui Settings](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/miui12_en_3.jpeg)

In the *Lock apps* screen, set the toggle switch for the AdGuard app to On.

![Miui Settings](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/miui12_en_4.jpeg)

That’s all! You’ve successfully pinned the AdGuard app. 

#### MIUI 12

![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomi4en.jpeg)

Go to Settings > Apps > Manage apps > AdGuard. 

- Set Autostart to “On”
- Set Battery saver to “No restrictions”

![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomi5en.jpeg)

Now launch the active apps manager by swiping up from the bottom of the screen and look for the AdGuard app. 

![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomi6.jpeg)

> Tap and hold on it until a special menu pops up. Select a lock icon.

Reviewer: hold it

![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomi7en.jpeg)

The lock icon should appear above the app window.

![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomi8en.jpeg)

> 

Reviewer: лишняя пустая строка

#### MIUI 10-11

To let your app run successfully in the background, configure settings as follows:

![Xiaomi](https://cdn.adguard.com/public/Adguard/screenshots/android/xiaomi1en.png)

- *Settings > Battery & performance* > switch-off *Battery saver* function

> 

Reviewer: лишняя пустая строка

![Xiaomi](https://cdn.adguard.com/public/Adguard/screenshots/android/xiaomi2en.png)

- Then open *App battery saver* settings > *AdGuard* > *No restrictions*

![Xiaomi](https://cdn.adguard.com/public/Adguard/screenshots/android/xiaomi3en.png)

> 

Reviewer: лишняя пустая строка

#### App battery saver

*Security > Battery > App Battery Saver > AdGuard > No restriction*

#### App pinning

To set up AdGuard's background work for Xiaomi devices you should pay attention to Battery and Permissions.

- Tap on the *Recent tasks* button and swipe AdGuard down to make options *visible* (as shown on the screenshot):

![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomirecent.png)

- Tap on the *lock* icon. This will stop Xiaomi from closing AdGuard automatically. It should look like this:

![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomilocked.png)

- Go to *"Battery"*

- Select *"battery saver"* app 

- Find and select AdGuard

- Set up the following *"Background settings"*:

![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomirest.png)

- Go to *"Permissions"*

- Select *"Autostart"*

- Make sure that autostart function is enabled for AdGuard:

![Xiaomi](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/xiaomiautostart.png)

## Samsung

#### Galaxy S9 / S10

> Battery optimization is **turned on by default**. At some unclear moments (maybe on app update, OS update), the settings also do revert back to the defaults, forcing you to turn them off again and again.

Reviewer: ```suggestion
Battery optimization is **turned on by default**. During the app update or the OS update, the settings also do revert back to the defaults, forcing you to turn them off again.
```

#### Sleeping apps

Sleeping apps menu is the sniper’s nest for Samsung’s app killing policies. Make sure to follow the instructions very carefully to prevent the apps from being killed.

Checklist:

- Disable **Put unused apps to sleep**
- Disable **Auto-disable unused apps**
- Remove your app from the list of sleeping apps
- Disable background restrictions for your app

1. Start **Device care** from phone settings

![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung1en.png)

2. Tap **Battery**

![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung2en.png)

> 

Reviewer: лишняя пустая строка

3. Tap the **3-dot menu** > **Settings**

![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung3en.png)

4. Disable all toggles (except Notifications)
5. Tap **Sleeping apps**

![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung45en.png)

6. **Wake up** all the apps using the trashcan icon

![Samsung](https://cdn.adguard.com/public/Adguard/screenshots/android/samsung6en.png)

> > **Note**: Make sure that Put unused apps to sleep and Auto-disable unused apps is disabled. Otherwise Samsung will put your apps back to sleep after a few days (3 by default) even if you have woken them up manually!

Reviewer: ```suggestion
> **Note**: Make sure that `Put unused apps to sleep` and `Auto-disable unused apps` are disabled. Otherwise, Samsung will put your apps back to sleep after a few days even if you have woken them up manually.
```

 
 ![Samsung settings](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/samsungoptions.png)

 - Tap on *Lock Apps*:
 
 ![Samsung settings](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/samsunglockapps.png)
 
  - Tap on the lock icon 
 
 ![Samsung settings](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/samsunglock.png)
 
## Huawei
 
#### App Launch on some EMUI 8, 9 and 10 devices (Huawei P20, Huawei P20 Lite, Huawei Mate 10…)

- *Phone settings* > *Battery* > *App launch* and then set your app to “Manage manually” and make sure everything is turned on.

![Huawei](https://cdn.adguard.com/public/Adguard/screenshots/android/huawei1en.png)

1. *Phone settings > Battery > App launch*. This feature may or may not be available for all devices or labeled differently.

![Huawei](https://cdn.adguard.com/public/Adguard/screenshots/android/huawei2en.png)

2. Turn off “Manage all automatically"

![Huawei](https://cdn.adguard.com/public/Adguard/screenshots/android/huawei3en.png)

> 

Reviewer: лишняя пустая строка (тут и ниже)

3. Make sure to **ENABLE** all toggles.

Also for reliable background processes you may need to uninstall PowerGenie as detailed below.

#### EMUI 9+ devices

> > Note: On some phones with EMUI 9+ (Android P+) Huawei introduced a new task killer app called PowerGenie which forces to quit all apps not whitelisted by Huawei and does not give users any configuration options. See below how to uninstall it.

Reviewer: ```suggestion
> Note: On phones with EMUI 9+ (Android P+) there is a task killer app called PowerGenie which forces to quit all apps not whitelisted by Huawei and does not give users any configuration options. See below how to uninstall it.

> Huawei is extremely inventive in breaking apps on their devices. In addition to all the non-standard power management measures described below, they introduced a new task killer app build right into EMUI 9 on Android Pie.

Reviewer: built

![Miui Settings](https://cdn.adguard.com/content/kb/ad_blocker/android/solving_problems/background-work/miui12_en_1.jpeg) 

> 

Reviewer: пустая строка
